packages/core/src/indexing-store/postgres/store.ts

```diff
@@ -449,36 +448,42 @@ export class PostgresIndexingStore implements IndexingStore {
   }) => {
     return this.wrap({ method: "update", tableName }, async () => {
       const table = `${tableName}_versioned`;
-      const formattedId = formatColumnValue({
-        value: id,
-        encodeBigInts: false,
-      });
+      const formattedId = encodeColumn(
+        id,
+        this.schema!.tables[tableName].id,
+        "postgres",
+      );
       const encodedCheckpoint = encodeCheckpoint(checkpoint);
 
       const row = await this.db.transaction().execute(async (tx) => {
         // Find the latest version of this instance.
         const latestRow = await tx
           .selectFrom(table)
           .selectAll()
-          .where(
-            "id",
-            this.idColumnComparator({ tableName, schema: this.schema }),
-            formattedId,
-          )
+          .where("id", "=", formattedId)
           .where("effectiveToCheckpoint", "=", "latest")
           .executeTakeFirstOrThrow();
 
         // If the user passed an update function, call it with the current instance.
-        let updateRow: ReturnType<typeof formatRow>;
+        let updateRow: ReturnType<typeof encodeRow>;
         if (typeof data === "function") {
-          const current = this.deserializeRow({
-            tableName,
-            row: latestRow,
-          });
+          const current = decodeRow(
+            latestRow,
+            this.schema!.tables[tableName],
```

Comment:
For the functions that use this many times, I'd consider doing a `const table = this.schema!.tables[tableName]` above.